### PR TITLE
multifile: add support for cache

### PR DESCRIPTION
**Pull request by mpranj**
Adds cache support for multifile (#2598).

## Basics

Check relevant points but **please do not remove entries**.
Do not describe the purpose of this PR in the PR description but:

- [x] Short descriptions should be in the release notes (added as entry in
      `doc/news/_preparation_next_release.md` which contains `_(my name)_`)
      **Please always add something to the the release notes.**
- [x] Longer descriptions should be in documentation or in design decisions.
- [x] Describe details of how you changed the code in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, should be in the commit messages.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests
- [x] I ran all tests locally and everything went fine
- [ ] affected documentation is fixed
- [ ] I added code comments, logging, and assertions (see doc/CODING.md)
- [ ] meta data is updated (e.g. README.md of plugins and doc/METADATA.md)

## Review

Remove the line below and add the "work in progress" label if you do
not want the PR to be reviewed:

### CODE REVIEW

**Comment by markus2330** [comment]
Thank you for working on this!

File: src/plugins/multifile/multifile.c [hunk 1]
```diff
@@ -53,6 +53,7 @@ typedef enum
 	ERROR = -1,
 	NOUPDATE = 0,
 	SUCCESS = 1,
+	CACHE_HIT = 2,
```
Comment (markus2330) [inline src/plugins/multifile/multifile.c]: Isn't a single definition of this enum better? (empty seems to be yet another error, I do not think we need this.)

File: src/plugins/multifile/multifile.c [hunk 2]
```diff
@@ -102,6 +103,9 @@ static inline Codes rvToRc (int rc)
 	case 1:
 		return SUCCESS;
 		break;
+	case 2:
+		return CACHE_HIT;
+		break;
```
Comment (markus2330) [inline src/plugins/multifile/multifile.c]: Isn't this whole function just a `return rc`? (Assuming rc is valid. The other cases should be asserted anyway.)

File: src/plugins/multifile/multifile.c (hunk 1 above)
Comment (mpranj) [reply src/plugins/multifile/multifile.c]: > Isn't a single definition of this enum better?

You mean for all plugins in `kdbplugin.h`? I think this is actually internal state to multifile and does not necessarily match the plugin return values.

> (empty seems to be yet another error, I do not think we need this.)

The multifile resolver definitely uses the EMPTY status, to remove empty files on commit.

File: src/plugins/multifile/multifile.c (hunk 2 above)
Comment (mpranj) [reply src/plugins/multifile/multifile.c]: It seems so ...

File: src/plugins/multifile/multifile.c (hunk 1 above)
Comment (markus2330) [reply src/plugins/multifile/multifile.c]: Ok, then better leave it as is.

File: src/plugins/multifile/multifile.c (hunk 2 above)
Comment (markus2330) [reply src/plugins/multifile/multifile.c]: Ok.

**Comment by mpranj** [comment]
@markus2330 there might be room for improvement in multifile, but I think for now we can merge it to get cache support.

**Comment by markus2330** [comment]
Thank you!

**Pull request merged**